File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.js

var maxProfit = function(prices) {
        let buy = prices[0]
    let profit = 0
    for(sell of prices){
        if(sell<buy){
            buy = sell
        }else{
            if(profit < sell-buy){
                profit = sell-buy
            }
        }
    }
    return profit
};

File: 0121-best-time-to-buy-and-sell-stock/0121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit= 0
        mini = prices[0]    
        for i in prices:
            if i < mini:
                mini=i
            profit= max(profit,(i - mini))
        return (profit) 
        
    
       